src/jsx.d.ts

```diff
@@ -1320,9 +1326,315 @@ export namespace JSXInternal {
 		onTransitionEndCapture?: TransitionEventHandler<Target>;
 	}
 
+	// All the WAI-ARIA 1.1 attributes from https://www.w3.org/TR/wai-aria-1.1/
+	export interface AriaAttributes {
+		/** Identifies the currently active element when DOM focus is on a composite widget, textbox, group, or application. */
+		'aria-activedescendant'?: Signalish<string | undefined>;
+		/** Indicates whether assistive technologies will present all, or only parts of, the changed region based on the change notifications defined by the aria-relevant attribute. */
+		'aria-atomic'?: Signalish<Booleanish | undefined>;
+		/**
+		 * Indicates whether inputting text could trigger display of one or more predictions of the user's intended value for an input and specifies how predictions would be
+		 * presented if they are made.
+		 */
+		'aria-autocomplete'?: Signalish<
+			'none' | 'inline' | 'list' | 'both' | undefined
+		>;
+		/** Indicates an element is being modified and that assistive technologies MAY want to wait until the modifications are complete before exposing them to the user. */
+		'aria-busy'?: Signalish<Booleanish | undefined>;
+		/**
+		 * Indicates the current "checked" state of checkboxes, radio buttons, and other widgets.
+		 * @see aria-pressed @see aria-selected.
+		 */
+		'aria-checked'?: Signalish<
+			boolean | 'false' | 'mixed' | 'true' | undefined
```

Comment:
> Meta question: should Booleanish exist? Not just the type, but the concept overall?

IMO, it probably should. It's inevitable that some user(s) will trip up on that and ask why both aren't supported. I think the ship has long since sailed on trying to enforce consistency w/ attributes.

Reply:
Oh thanks! Yes! I meant to go back and update this but forgot.

Yea, the handling of boolean type vs string is interesting/complicated. In React, for some properties `false` removes the attributes but for others (e.g. ARIA) `false` is stringified. Personally, I'd prefer to just always say pass the string `"false"` if that's the value the attribute should have but we should match React's behavior here and accept both booleans and strings. There is some nice ergonomics here too.

If you are curious, here is a list of the different attribute behaviors React special cases: https://github.com/facebook/react/blob/c04b18070145b82111e1162729f4776f4d2c6112/packages/react-dom-bindings/src/shared/DOMProperty.js#L18-L49